cmd/bundle/init.go

```diff
@@ -196,7 +198,26 @@ See https://docs.databricks.com/en/dev-tools/bundles/templates.html for more inf
 	cmd.Flags().StringVar(&branch, "tag", "", "Git tag to use for template initialization")
 	cmd.Flags().StringVar(&tag, "branch", "", "Git branch to use for template initialization")
 
-	cmd.PreRunE = root.MustWorkspaceClient
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		// Configure the logger to send telemetry to Databricks.
+		ctx := telemetry.ContextWithLogger(cmd.Context())
+		cmd.SetContext(ctx)
+
+		return root.MustWorkspaceClient(cmd, args)
+	}
+
+	cmd.PostRun = func(cmd *cobra.Command, args []string) {
+		ctx := cmd.Context()
+		w := root.WorkspaceClient(ctx)
+		apiClient, err := client.New(w.Config)
+		if err != nil {
+			// Uploading telemetry is best effort. Do not error.
+			return
+		}
+
+		telemetry.Flush(cmd.Context(), apiClient)
```

Comment:
Shall we log any errors occuring?

Reply:
If you mean errors during `bundle init`, that would have to be a separate PR since we'd need to ensure we do not log PII in the error messages. Ideally thus we'll introduce some sort of structured error messages for the subset of errors we want telemetry for.

If you mean logging to stderr, good point, I'll do that.